tests/unit/restapi/v1/test_groups.py

```diff
@@ -0,0 +1,373 @@
+
+    if paging_info is not None:
+        query_string["index"] = paging_info["index"]
+        query_string["pageLength"] = paging_info["page_length"]
+
+    response = client.get(
+        f"/{V1_ROOT}/{V1_GROUPS_ROUTE}",
+        query_string=query_string,
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["data"] == expected
+
+
+def assert_registering_existing_group_name_fails(
+    client: FlaskClient, name: str
+) -> None:
+    """Assert that registering a group with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        name: The name to assign to the new group.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = actions.register_group(client, name=name)
+    assert response.status_code == 400
+
+
+def assert_group_name_matches_expected_name(
+    client: FlaskClient, group_id: int, expected_name: str
+) -> None:
+    """Assert that the name of a group matches the expected name.
+
+    Args:
+        client: The Flask test client.
+        group_id: The id of the group to retrieve.
+        expected_name: The expected name of the group.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the name of the
+            group does not match the expected name.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_GROUPS_ROUTE}/{group_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_cannot_rename_group_with_existing_name(
+    client: FlaskClient,
+    group_id: int,
+    existing_name: str,
+) -> None:
+    """Assert that renaming a group with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        group_id: The id of the group to rename.
+        name: The name of an existing group.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = modify_group(
+        client=client,
+        group_id=group_id,
+        new_name=existing_name,
+    )
+    assert response.status_code == 400
+
+
+# -- Tests -----------------------------------------------------------------------------
+
+
+@pytest.mark.v1
+def test_create_group(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+) -> None:
+    """Test that groups can be correctly registered and retrieved using the API.
+
+    Given an authenticated user, this test validates the following sequence of actions:
+
+    - The user registers a group named "new_group".
+    - The response is valid matches the expected values given the registration request.
+    - The user is able to retrieve information about the group using the group id^[1].
+
+    [1] The group id is generated in the backend during the POST and is returned in the response.
+    """
+    name = "new_group"
+    user_id = auth_account["user_id"]
+
```

Comment:
nitpick: remove whitespace

Reply:
Removed in commit: [0d7f47c](https://github.com/usnistgov/dioptra/pull/458/commits/0d7f47c2b08459889a5064a463670fa441bfcca1)